web/blog/2022-06-24-ML-code-gen-vs-coding-by-hand-future.md

```diff
@@ -0,0 +1,243 @@
+
+We don’t need to care what exact packages or encryption methods are used beneath the hood - it is the responsibility we trust with the authors and maintainers of the abstraction layer, just like we trust that Python knows the best how to sum two numbers on the assembly level and that it is kept in sync with the latest advancements in the field. The same happens when we rely on the built-in data structures or count on the garbage collector to manage our program’s memory well.
+
+## But my beautiful generated codez 😿💻! What happens with it then?
+
+Don’t worry, it’s all still here and you can generate all the code you wish! The main point to understand here is that ML code generation and framework/language development complement rather than replace each other and are here to stay, which is ultimately a huge win for the developer community - they will keep making our lives easier and allow us to do more fun stuff (instead of implementing auth or CRUD API for the n-th time)!
+
+I see the evolution here as a cycle (or an upward spiral in fact, but that’s beyond my drawing capabilities):
+
+1. **language/framework exists**, is mainstream, and a lot of people use it
+2. **patterns start emerging** (e.g. implementing auth, or making an API call) → ML captures them, offers via autocomplete
+3. **some of those patterns mature** and become stable → candidates for abstraction
+4. **new, more abstract, language/framework** emerges
+5. **back to step 1**.
+
+<ImgWithCaption
+    alt="Language evolution lifecycle"
+    source="img/language-lifecycle-copilot.png"
+    caption="It’s the circle of (language) life, and it moves us all - Ingonyama nengw' enamabala, …"
+/>
+
+## Conclusion
+
+This means we are winning on both sides - when the language is mainstream we can benefit from ML code generation, helping us write the code faster. On the other hand, when the patterns of code we don’t want to repeat/deal with emerge and become stable we get a whole new language or framework that allows us to write even less code and care about fewer implementation details!
+
+<ImgWithCaption
+    alt="Fizz Buzz with Copilot - stop"
+    source="img/fizz-buzz-copilot-stop.gif"
+    caption="The future is now, old man."
+/>
+
+
+\**Not to be biased, there are also other solutions offering similar functionality - e.g. [TabNine](https://www.tabnine.com/), [Webstorm has its own](https://www.jetbrains.com/help/webstorm/auto-completing-code.html#ml_completion), [Kite](https://www.kite.com/), [GPT Code Clippy (OSS attempt)](https://github.com/CodedotAl/gpt-code-clippy) et al., but Github Copilot recently made the biggest splash.*
+
+### Writing that informed this post
```

Comment:
@shayneczyzewski here it is! not sure why it hasn't showed up for you.

Reply:
Great, I see it now after doing a force reset to remote head! Looking good! 👍🏻 Lots of reviewers, I like it :D